examples/vault-iam-auth/README.md

```diff
@@ -57,28 +50,48 @@ of the Vault nodes.
 
 ### Vault Authentication using IAM user or role
 
-To read more about Vault IAM auth, refer to [Vault AWS Auth documentation][aws_auth].
+IAM auth is a process in which Vault leverages on AWS STS (Security Token Service) to
+identify the AWS IAM principal (user or role) attached to an AWS resource such as
+an ECS Task or a Lambda Function that originates the login request. You can still
+use the `iam` method for EC2 instances attached to a role, like we do in this example,
+but for a login method specifically for EC2 instances, please refer to the
+[`ec2` auth method example][ec2_example].
+
+The workflow is that the client trying to authenticate will create a request to
+the method `GetCallerIdentity` of the AWS STS API. This method basically answers
+the question "Who am I?". This request is then signed with the AWS credentials of
+the client. The signed result is then sent with the login request to the Vault Server.
+When the Vault server receives a login request with the `iam` method, it can execute
+the STS request without actually knowing the contents of the signed part. It then
+receives a response from STS identifying who signed it, which the Vault Server then
+can check against the ARN of the IAM principal bounded to a previously created Vault
+Role and decide if it should be allowed to authenticate or not.
```

Comment:
Maybe because it's not my first time reading it, but I find your explanation clearer than the Vault docs 👍